qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -83,7 +81,7 @@ class VQE(VQAlgorithm, MinimumEigensolver):
 
     def __init__(self,
                  operator: Optional[BaseOperator] = None,
-                 var_form: Optional[VariationalForm] = None,
+                 var_form: Optional[Union[QuantumCircuit, VariationalForm]] = None,
```

Comment:
How much are we going to have backward compatibility here. The order of params was preserved even after making Operator optional. Neither what is called var form or optimizer should ideally be optional as they are both required. I am not sure ow many people used the names since prior to this release they were not optional params but usually supplied by order. Though is one did it would be a breaking change if it were eliminated. Its possible of course to rename it here and add it at the end of the params as the old name but raise a deprecated warning should it be used.

Reply:
We're using the `@deprecate_arguments` decorator a lot in Terra already and I used them for the N-local business to keep backward compatibility, which does exactly what you proposed in the last sentence. 👍